src/modes/draw_line_string.js

```diff
@@ -1,96 +1,101 @@
 const CommonSelectors = require('../lib/common_selectors');
 const LineString = require('../feature_types/line_string');
 const isEventAtCoordinates = require('../lib/is_event_at_coordinates');
-const doubleClickZoom = require('../lib/double_click_zoom');
 const Constants = require('../constants');
 const createVertex = require('../lib/create_vertex');
 
-module.exports = function(ctx) {
-  const line = new LineString(ctx, {
-    type: Constants.geojsonTypes.FEATURE,
-    properties: {},
-    geometry: {
-      type: Constants.geojsonTypes.LINE_STRING,
-      coordinates: []
+import ModeInterface from './mode_interface';
+
+export default class DrawLineStringMode extends ModeInterface {
+  constructor(store, ui) {
```

Comment:
Using an object, then, would allow you to delete an argument without changing the signature (e.g. if we decided to no longer pass the first argument into a method).

> I think we can handle your concerns about exposing the internals by adding more default functions to the mode like this.setSelected to let the mode set the selected features.

I think that is a good idea. Are you thinking we'd then *not* pass the arguments to custom modes, and expect them to rely on the class's methods?